code/crates/engine/src/sync.rs

```diff
@@ -296,6 +298,33 @@ where
                         .success_or(eyre!("Failed to get decided value for height {height}"))?;
 
                     if let Some(value) = value {
+                        let value_response = ValueResponse(sync::ValueResponse::new(
+                            *range.start(),
+                            vec![value.clone()],
+                        ));
+
+                        let result = ractor::call!(self.gossip, move |reply_to| {
+                            NetworkMsg::GetResponseSize(value_response.clone(), reply_to)
+                        });
+
+                        let total_value_size_bytes = match result {
+                            Ok(value_in_bytes) => value_in_bytes,
+                            Err(e) => {
+                                error!("Failed to get response size for value, stopping at for height {}: {:?}", height, e);
+                                break;
+                            }
+                        };
+
+                        // check if adding this value would exceed the max-response limit
+                        if response_size_bytes + total_value_size_bytes
+                            > self.sync_config.max_response_size
```

Comment:
Deleted my previous suggestion to change the check because I realize that it would break early. But if it is equal then we should push the value and break, not go another round and only then break. So a separate 

So you add a check below line 328:

```
if response_size_bytes + total_value_size_bytes == self.sync_config.max_response_size {

break;

 }
```

Reply:
Seems like a quite unlikely scenario. In any case, I added an extra check.